xai_components/xai_utils/utils.py

```diff
@@ -709,3 +767,5 @@ class AwaitFutures(Component):
     def execute(self, ctx) -> None:
         from concurrent.futures import wait
         wait(self.futures.value)
+        LOGGER.info(f'Parallel execution finished. For more details check the log file: {LOG_FILE_PATH}')
```

Comment:
I expect we could take the log file path directly from the LOGGER object. If this is the case, we could avoid the need to define it globally in builder.py

Reply:
For the ForEach explanation I could add a comment in the xircuits file like this:
![image](https://github.com/user-attachments/assets/46e2a23c-ebe1-45fd-ac2f-8b03ff1a46cd)
what do you think?

And for the logger, I can get the file path in the AwaitFutures component like this:
`log_file_path = LOGGER.parent.handlers[1].baseFilename`
other option would be to define it as an instance attribute for the logger object, but I like the first option better.. what do you think?